sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnostics.java

```diff
@@ -34,8 +35,8 @@
     static final String USER_AGENT = Utils.getUserAgent();
     static final String USER_AGENT_KEY = "userAgent";
 
-    CosmosDiagnostics(DiagnosticsClientContext diagnosticsClientContext) {
-        this.clientSideRequestStatistics = new ClientSideRequestStatistics(diagnosticsClientContext);
+    CosmosDiagnostics(DiagnosticsClientContext diagnosticsClientContext, GlobalEndpointManager globalEndpointManager) {
```

Comment:
not blocking comment (just a suggestion): 
ideally CosmosDiagnostics should only have read access to the `GlobalEndpoingManager` we should define an interface for `GlobalEndpointManager` which only expose the read api needed.